# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/flyrank-bih/flyrank-ml-internship-starter/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

I'm choosing Lane 2 — Refresh / Content Opportunity Scoring. This lane looks at which existing pages should be reviewed first for refresh, based on signs of decline, staleness, or missed opportunity. I picked it because it has a clear, real-world decision behind it, and the starter pipeline I already ran was built for exactly this lane.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

Question: Which pages should be reviewed first for refresh, based on signs of decline, staleness, or missed opportunity?

Unit of analysis: one page (one row = one content page)
Decision it improves: which pages a content reviewer spends their limited time checking first
Action taken: update, expand, or leave-and-monitor a flagged page
Cost of a wrong call: if the wrong pages are flagged, reviewers waste time on pages that didn't need help, while pages that are actually declining go unnoticed longer
Why data/ML helps: with hundreds of thousands of pages, no one can check them all by hand — a simple fixed rule misses how signals combine, but a scoring model can weigh several signals together to find the pages that matter most

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

A quick look at the starter dataset shows real evidence that this lane is worth pursuing — see the numbers below.

In [9]:
!git clone https://github.com/sarahsaeed564/Machine_Learning.git
%cd Machine_Learning

Cloning into 'Machine_Learning'...
remote: Enumerating objects: 129, done.
remote: Counting objects: 100% (129/129), done.
remote: Compressing objects: 100% (83/83), done.
remote: Total 129 (delta 44), reused 97 (delta 30), pack-reused 0 (from 0)
Receiving objects: 100% (129/129), 1.84 MiB | 11.95 MiB/s, done.
Resolving deltas: 100% (44/44), done.
/content/Machine_Learning/Machine_Learning/Machine_Learning/Machine_Learning


In [10]:
import pandas as pd

df = pd.read_csv("data/raw/content_refresh_anonymized.csv")

print("Total pages:", len(df))
print("Pages flagged declining:", (df["trend_direction"] == "down").sum())
print("Percent declining:", round((df["trend_direction"] == "down").mean() * 100, 1), "%")
print("Median impressions in last 90 days:", df["impressions_90d"].median())

Total pages: 30000
Pages flagged declining: 16262
Percent declining: 54.2 %
Median impressions in last 90 days: 731.0


Out of 30,000 pages in the starter dataset, 54.2% (16,262 pages) show a declining trend, with a median of 731 impressions in the last 90 days — showing a large, high-traffic pool of pages that would genuinely benefit from a prioritized review process rather than manual checking.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

This data shows patterns I observed, not proof of cause and effect. A page flagged as declining is not guaranteed to recover if refreshed — showing that would require a controlled experiment. My output is meant to help a human reviewer prioritize pages, not to guarantee results

## 5.Self-check

Lane picked: Lane 2 — Refresh / Content Opportunity Scoring, with a clear reason why.
Decision and action named: which pages a reviewer checks first; the action is to update, expand, or monitor a flagged page.
Real numbers shown: 30,000 total pages, 54.2% flagged declining, median 731 impressions in the last 90 days.
Why this isn't just "train a model": the value is in defining a clear, defensible question and evidence base first — a model only matters if it beats a simple rule at ranking pages correctly, which hasn't been tested yet at this stage.
Careful language used: results are described as observed patterns, not proof that a refresh causes recovery; the output supports human review, it doesn't replace it.